src/javascript/ContentEditor/editorTabs/EditPanelContent/Preview/PreviewViewers/IframeViewer/IframeViewer.jsx

```diff
@@ -49,61 +49,53 @@ function loadAssets(assets, iframeDocument) {
     const iframeRef = useRef(null);
+    const onLoadTimeoutRef = useRef(null);
+    let displayValue = data && data.nodeByPath && data.nodeByPath.renderedContent ? data.nodeByPath.renderedContent.output : '';
+    if (displayValue === '') {
+        displayValue = t('label.contentManager.contentPreview.noViewAvailable');
+    }
 
-    const renderIFrame = useCallback(() => {
-        const element = iframeRef.current;
-
-        if (!element) {
-            return;
-        }
-
+    useEffect(() => {
         setLoading(true);
-        let displayValue = data && data.nodeByPath && data.nodeByPath.renderedContent ? data.nodeByPath.renderedContent.output : '';
-        if (displayValue === '') {
-            displayValue = t('label.contentManager.contentPreview.noViewAvailable');
-        }
-
-        const iframeWindow = element.contentWindow || element;
-        writeInIframe(displayValue, iframeWindow)
-            .then(() => {
-                iframeWindow.document.body.setAttribute('style', 'pointer-events: none');
-            })
-            .then(() => {
+        // Add a timer to always remove the loader and click on links after 5 seconds
+        onLoadTimeoutRef.current = setTimeout(() => {
+            console.debug('iframe onLoad did not trigger, remove loader and disable click events');
+            const element = iframeRef.current;
+            const iframeWindow = element.contentWindow || element;
+            iframeWindow?.document?.body?.setAttribute('style', 'pointer-events: none');
+            setLoading(false);
+        }, 5000);
+        return () => clearTimeout(onLoadTimeoutRef.current);
+    }, [displayValue]);
+
+    const onLoad = () => {
+        try {
+            console.debug('Preview loaded, add assets and remove loader');
+            const element = iframeRef.current;
+            const iframeWindow = element.contentWindow || element;
+            iframeWindow.document.body.setAttribute('style', 'pointer-events: none');
+
+            if (previewContext.contextConfiguration !== 'page') {
                 const assets = data && data.nodeByPath && data.nodeByPath.renderedContent ?
-                    data.nodeByPath.renderedContent.staticAssets :
-                    [];
-                return loadAssets(assets, iframeWindow.document);
-            })
-            .catch(err => console.error('Error in the preview', err))
-            .then(() => {
-                // No zoom on full if no content wrapped in the page
-                if (previewContext.requestAttributes) {
-                    zoom(iframeWindow.document, onContentNotFound, editorContext);
-                }
-            })
-            .then(() => {
-                setLoading(false);
-            });
-    }, [data, editorContext, onContentNotFound, previewContext.requestAttributes, t]);
+                    data.nodeByPath.renderedContent.staticAssets : [];
+                loadAssets(assets, iframeWindow.document);
```

Comment:
I see that loadAssets is async. does the timing of the `setLoading(false)` matters here i.e. do we need to clear the loading inside a `then` clause for this case instead?

Reply:
Discussion with @dgriffon leaving as-is for now as it's better to cancel loader earlier with no breaking side effects.